src/manager.cpp

```diff
@@ -212,12 +212,91 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 int Manager::updateKeyword(const types::Path i_vpdPath,
                            const types::WriteVpdParams i_paramsToWriteData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_vpdPath;
-    (void)i_paramsToWriteData;
+    if (i_vpdPath.empty())
+    {
+        logging::logMessage("Given VPD path is empty.");
+        return -1;
+    }
 
-    // On success return number of bytes written. On failure return -1.
-    return -1;
+    nlohmann::json l_sysCfgJsonObj{};
+
+    if (m_worker.get() != nullptr)
+    {
+        l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+    }
+
+    int l_bytesUpdated = -1;
+    types::Path l_hwPath = i_vpdPath;
+    types::Path l_redundantFruPath;
+
+    try
+    {
+        if (!l_sysCfgJsonObj.empty())
+        {
+            // Get hardware path from system config JSON.
+            const std::string& l_fruPath =
+                jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_vpdPath);
+
+            if (!l_fruPath.empty())
+            {
+                // Save the FRU path to update on hardware
+                l_hwPath = l_fruPath;
+
+                // Get inventory object path from system config JSON
+                const std::string& l_inventoryObjPath =
+                    jsonUtility::getInventoryObjPathFromJson(l_sysCfgJsonObj,
+                                                             i_vpdPath);
+
+                if (!l_inventoryObjPath.empty())
+                {
+                    // TODO: Extract the record, keyword, and value from input
```

Comment:
Single TODD ,  call write on inventory path

Reply:
We may not have a single api which does all these todo steps.
Some pieces of code (like extract from input, size comparisons) may require to be in this api only.

so having in separate lines makes easier to understand the steps and segregate the pieces when required

Reply:
then would it be better to write the hardware first , and then read the hardware and write the Dbus later ?

Reply:
Just that I am not fully convinced by this size check